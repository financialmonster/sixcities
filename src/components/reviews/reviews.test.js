import React from 'react';
import renderer from 'react-test-renderer';

import Reviews from './reviews.jsx';

const reviews = [{id: 1, rating: 3, date: `2020-04-13T20:25:41.450Z`, user: {id: 15, is_pro: false, name: `Kendall`, 
    avatar_url: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/6.jpg`},
    comment: `I stayed here for one night and it was an unpleasant experience.`},
    {id: 2, rating: 2, date: `2020-04-13T20:25:41.450Z`, user: {id: 13, is_pro: false, name: `Zak`,
    avatar_url: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/4.jpg`},
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an
    amazing view of the river!`}
];

it(`Correctly renders the Reviews component`, () => {
    const tree = renderer.create(<Reviews reviews={reviews} />).toJSON();

    expect(tree).toMatchSnapshot();
});
