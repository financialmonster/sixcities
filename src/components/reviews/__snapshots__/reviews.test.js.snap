// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders the Reviews component 1`] = `
Array [
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/6.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Kendall
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        I stayed here for one night and it was an unpleasant experience.
      </p>
      <time
        className="reviews__time"
        dateTime="2020-4-14"
      >
        April 2020
      </time>
    </div>
  </li>,
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/4.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Zak
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        What an amazing view! The house is stunning and in an amazing location. The large glass wall had an
    amazing view of the river!
      </p>
      <time
        className="reviews__time"
        dateTime="2020-4-14"
      >
        April 2020
      </time>
    </div>
  </li>,
]
`;
